addons/web_editor/static/src/js/widgets/widgets.js

```diff
@@ -550,6 +573,93 @@ var ImageWidget = MediaWidget.extend({
         });
         this.$el.submit();
     },
+    /**
+     * @private
+     */
+    _loadMoreImages: function () {
+        this.IMAGES_ROWS = this.IMAGES_ROWS + 2;
+        this.IMAGES_PER_PAGE = this.IMAGES_ROWS * this.IMAGES_PER_ROW;
+        this._renderImages();
+        this._displayLoadMoreBtn();
+    },/**
+     * @private
+     */
+    _loadMoreUnsplash: function () {
+        this.IMAGES_UNSPLASH_ROWS = this.IMAGES_UNSPLASH_ROWS + 2;
+        this.IMAGES_UNSPLASH_PER_PAGE = this.IMAGES_UNSPLASH_ROWS * this.IMAGES_UNSPLASH_PER_ROW;
+        this._renderImages();
+    },
+    /**
+     * @private
+     */
+     _displayLoadMoreBtn: function () {
+        var records = this.records.length;
+        var loadMoreBtn = this.$el.find('.o_load_more_images');
+        var parentDiv = this.$el.find('.o_load_elements');
+        if (this.IMAGES_PER_PAGE >= records) {
+            loadMoreBtn.hide();
+            parentDiv.append(QWeb.render('web_editor.dialog.image.loaded.content'));
+        }
+    },
+    /**
+     * @private
+     */
+    _checkImageUrl: function () {
+        var format = ['.gif','.jpe','.jpg','.png'];
+        var url = this.$('#imageurl').val();
+        var acceptedURL = false;
+        var warning = this.$('.warning-url-image');
+        var success = this.$('.success-url-image');
+
+        function nameButton(name) {
+            if (name === 'document') {
+                name = "Add a document URL";
+            } else {
+                name = "Add an image URL";
+            }
+            $('.o_upload_image_url_button').html(name);
+        }
+
+        for (var i in format) {
+            var endsWith = url.endsWith(format[i]);
+            if (endsWith) {
+                acceptedURL = true;
+                warning.addClass('d-none');
+                success.removeClass('d-none');
+                nameButton('image');
+            }
+        }
+        if (!acceptedURL) {
+            warning.removeClass('d-none');
+            success.addClass('d-none');
+            nameButton('document');
+        }
+        if (url.length < 1) {
+            warning.addClass('d-none');
+            success.addClass('d-none');
+            nameButton('image');
+        }
+    },
+    /**
+     * @private
+     */
+    _checkDocumentUrl: function () {
+        var url = this.$('#documenturl').val();
+        var error = this.$('.error-url-document');
+        var success = this.$('.success-url-document');
+        var regex = /^(http:\/\/www\.|https:\/\/www\.|http:\/\/|https:\/\/)?[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?$/;
+        if (url.match(regex) !== null) {
+            success.removeClass('d-none');
+            error.addClass('d-none');
+        } else {
+            success.addClass('d-none');
+            error.removeClass('d-none');
+        }
```

Comment:
That sort of stuff can be done like this for more readability:
```javascript
var success = (url.match(regex) !== null);
$success.toggleClass('d-none', !success);
$error.toggleClass('d-none', success);
```
Also see that every jQuery element variables must start with a `$`